File: frontend/src/api/bo/Comment.js
import BusinessObject from './BusinessObject';


export default class Comment extends BusinessObject {
    constructor(text, person_id, card_id) {
        super();
        this.text = text;
        this.person_id = person_id;
        this.card_id = card_id;
    }

    setText(text) {
        this.text = text;
    }

    getText() {
        return this.text;
    }

    setPersonId(person_id) {
        this.person_id = person_id;
    }

    getPersonId() {
        return this.person_id;
    }

    setCardId(card_id) {
        this.card_id = card_id;
    }

    getCardId() {
        return this.card_id;
    }

    static fromJSON(comment) {
        let result = [];
        if (Array.isArray(comment)) {
            comment.forEach((c) => {
                Object.setPrototypeOf(c, Comment.prototype);
                result.push(c);
            })
        } else {
            let c = comment;
            Object.setPrototypeOf(c, Comment.prototype);
            result.push(c);
        }
        return result;
    }
}

File: src/server/bo/comment.py
from server.bo.business_object import BusinessObject


class Comment(BusinessObject):
    def __init__(self):
        super().__init__()
        self._text = ""
        self._person_id = 0
        self._card_id = 0

    def get_text(self):
        return self._text

    def set_text(self, text):
        self._text = text

    def get_person_id(self):
        return self._person_id

    def set_person_id(self, person_id):
        self._person_id = person_id

    def get_card_id(self):
        return self._card_id

    def set_card_id(self, card_id):
        self._card_id = card_id

    @staticmethod
    def from_dict(comment_dict=dict()):
        comment = Comment()
        comment.set_id(comment_dict.get("id", 0))
        comment.set_created_at(Comment.time_format(comment_dict.get("created_at", None)))
        comment.set_last_update_at(Comment.time_format(comment_dict.get("last_update_at", None)))
        comment.set_text(comment_dict.get("text", ""))
        comment.set_person_id(comment_dict.get("person_id", 0))
        comment.set_card_id(comment_dict.get("card_id", 0))
        return comment